tools/serverpod_cli/lib/src/analyzer/models/definitions.dart

```diff
@@ -89,15 +89,31 @@ class ClassDefinition extends SerializableModelDefinition {
         .firstWhere((element) => element.name == name, orElse: () => null);
   }
 
+  /// Returns the `ClassDefinition` of the parent class.
+  /// If there is no parent class, `null` is returned.
   ClassDefinition? get parentClass {
     var extendsClass = this.extendsClass;
     if (extendsClass is! ResolvedInheritanceDefinition) return null;
 
     return extendsClass.classDefinition;
   }
 
-  List<SerializableModelFieldDefinition> get parentFields =>
-      parentClass?.fields ?? [];
+  /// Returns a list of all fields in the parent class.
+  /// If there is no parent class, an empty list is returned.
+  List<SerializableModelFieldDefinition> get inheritedFields =>
+      parentClass?.fieldsIncludingInherited ?? [];
+
+  /// Returns a list of all fields in this class, including inherited fields.
+  /// It ensures that the 'id' field, if present, is always included at the beginning of the list.
+  List<SerializableModelFieldDefinition> get fieldsIncludingInherited {
```

Comment:
Question: Should this really be `fields` and then `fields` should be `modelDefinedFields`?

Reply:
(This can always be changed later so not a blocker for this PR)

Reply:
cool I'll make a small PR for that change then :)